cvs/cvs/doc/cvs.1
.\" This is the man page for CVS.  It is auto-generated from the
.\" cvs.man.header, cvs.texinfo, & cvs.man.footer files.  Please make changes
.\" there.  A full copyright & license notice may also be found in cvs.texinfo.
.\"
.\" Man page autogeneration, including this header file, is
.\" Copyright 2004-2005 The Free Software Foundation, Inc.,
.\"                     Derek R. Price, & Ximbiot <http://ximbiot.com>.
.\"
.\" This documentation is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2, or (at your option)
.\" any later version.
.\"
.\" This documentation is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License
.\" along with this documentation; if not, write to the Free Software
.\" Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
.de Id
.ds Rv \\$3
.ds Dt \\$4
..
.TH CVS 1 "\*(Dt"
.\" Full space in nroff; half space in troff
.de SP
.if n .sp
.if t .sp .5
..
.\" quoted command
.de `
.RB ` "\|\\$1\|" '\\$2
..
.SH "NAME"
cvs \- Concurrent Versions System
.SH "SYNOPSIS"
.TP
\fBcvs\fP [ \fIcvs_options\fP ]
.I cvs_command
[
.I command_options
] [
.I command_args
]
.SH "NOTE"
.IX "revision control system" "\fLcvs\fR"
.IX  cvs  ""  "\fLcvs\fP \- concurrent versions system"
.IX  "concurrent versions system \- \fLcvs\fP"
.IX  "release control system"  "cvs command"  ""  "\fLcvs\fP \- concurrent versions system"
.IX  "source control system"  "cvs command"  ""  "\fLcvs\fP \- concurrent versions system"
.IX  revisions  "cvs command"  ""  "\fLcvs\fP \- source control"
This manpage is a summary of some of the features of
\fBcvs\fP.  It is auto-generated from an appendix of the CVS manual.
For more in-depth documentation, please consult the
Cederqvist manual (via the
.B info CVS
command or otherwise,
as described in the SEE ALSO section of this manpage).  Cross-references
in this man page refer to nodes in the same.
.SH "CVS commands"
.SS "Guide to CVS commands"
.SP
This appendix describes the overall structure of
\fBcvs\fR commands, and describes some commands in
detail (others are described elsewhere; for a quick
reference to \fBcvs\fR commands, see node `Invoking CVS\(aq in the CVS manual).
.SP
.SH "Structure"
.SS "Overall structure of CVS commands"
.IX "Structure"
.IX "CVS command structure"
.IX "Command structure"
.IX "Format of CVS commands"
.SP
The overall format of all \fBcvs\fR commands is:
.SP
.PD 0
.SP
.IP "" 2
cvs [ cvs_options ] cvs_command [ command_options ] [ command_args ]

.PD
.IP "" 0
.SP
.IP "" 0
\fBcvs\fR
.IP "" 2
The name of the \fBcvs\fR program.
.SP
.IP "" 0
\fBcvs_options\fR
.IP "" 2
Some options that affect all sub-commands of \fBcvs\fR.  These are
described below.
.SP
.IP "" 0
\fBcvs_command\fR
.IP "" 2
One of several different sub-commands.  Some of the commands have
aliases that can be used instead; those aliases are noted in the
reference manual for that command.  There are only two situations
where you may omit \fBcvs_command\fR: \fBcvs -H\fR elicits a
list of available commands, and \fBcvs -v\fR displays version
information on \fBcvs\fR itself.
.SP
.IP "" 0
\fBcommand_options\fR
.IP "" 2
Options that are specific for the command.
.SP
.IP "" 0
\fBcommand_args\fR
.IP "" 2
Arguments to the commands.
.SP
There is unfortunately some confusion between
\fBcvs_options\fR and \fBcommand_options\fR.
When given as a \fBcvs_option\fR, some options only
affect some of the commands.  When given as a
\fBcommand_option\fR it may have a different meaning, and
be accepted by more commands.  In other words, do not
take the above categorization too seriously.  Look at
the documentation instead.
.SP
.SH "Exit status"
.SS "CVS\(aqs exit status"
.IX "Exit status, of CVS"
.SP
\fBcvs\fR can indicate to the calling environment whether it
succeeded or failed by setting its \fIexit status\fR.
The exact way of testing the exit status will vary from
one operating system to another.  For example in a unix
shell script the \fB$?\fR variable will be 0 if the
last command returned a successful exit status, or
greater than 0 if the exit status indicated failure.
.SP
If \fBcvs\fR is successful, it returns a successful status;
if there is an error, it prints an error message and
returns a failure status.  The one exception to this is
the \fBcvs diff\fR command.  It will return a
successful status if it found no differences, or a
failure status if there were differences or if there
was an error.  Because this behavior provides no good
way to detect errors, in the future it is possible that
\fBcvs diff\fR will be changed to behave like the
other \fBcvs\fR commands.
.SP
.SH "~/.cvsrc"
.SS "Default options and the ~/.cvsrc file"
.IX "\&.cvsrc file"
.IX "Option defaults"
.SP
There are some \fBcommand_options\fR that are used so
often that you might have set up an alias or some other
means to make sure you always specify that option.  One
example (the one that drove the implementation of the
\fB.cvsrc\fR support, actually) is that many people find the
default output of the \fBdiff\fR command to be very
hard to read, and that either context diffs or unidiffs
are much easier to understand.
.SP
The \fB~/.cvsrc\fR file is a way that you can add
default options to \fBcvs_commands\fR within cvs,
instead of relying on aliases or other shell scripts.
.SP
The format of the \fB~/.cvsrc\fR file is simple.  The
file is searched for a line that begins with the same
name as the \fBcvs_command\fR being executed.  If a
match is found, then the remainder of the line is split
up (at whitespace characters) into separate options and
added to the command arguments \fIbefore\fR any
options from the command line.
.SP
If a command has two names (e.g., \fBcheckout\fR and
\fBco\fR), the official name, not necessarily the one
used on the command line, will be used to match against
the file.  So if this is the contents of the user\(aqs
\fB~/.cvsrc\fR file:
.SP
.PD 0
.SP
.IP "" 2
log -N
.IP "" 2
diff -uN
.IP "" 2
rdiff -u
.IP "" 2
update -Pd
.IP "" 2
checkout -P
.IP "" 2
release -d

.PD
.IP "" 0
.SP
the command \fBcvs checkout foo\fR would have the
\fB-P\fR option added to the arguments, as well as
\fBcvs co foo\fR.
.SP
With the example file above, the output from \fBcvs
diff foobar\fR will be in unidiff format.  \fBcvs diff
-c foobar\fR will provide context diffs, as usual.
Getting "old" format diffs would be slightly more
complicated, because \fBdiff\fR doesn\(aqt have an option
to specify use of the "old" format, so you would need
\fBcvs -f diff foobar\fR.
.SP
In place of the command name you can use \fBcvs\fR to
specify global options (see node `Global options\(aq in the CVS manual).  For
example the following line in \fB.cvsrc\fR
.SP
.PD 0
.SP
.IP "" 2
cvs -z6

.PD
.IP "" 0
.SP
causes \fBcvs\fR to use compression level 6.
.SP
.SH "Global options"
.IX "Options, global"
.IX "Global options"
.IX "Left-hand options"
.SP
The available \fBcvs_options\fR (that are given to the
left of \fBcvs_command\fR) are:
.SP
.IP "" 0
\fB--allow-root=\fIrootdir\fB\fR
.IP "" 2
May be invoked multiple times to specify one legal \fBcvsroot\fR directory with
each invocation.  Also causes CVS to preparse the configuration file for each
specified root, which can be useful when configuring write proxies,  See
see node `Password authentication server\(aq in the CVS manual & see node `Write proxies\(aq in the CVS manual.
.SP
.IX "Authentication, stream"
.IX "Stream authentication"
.IP "" 0
\fB-a\fR
.IP "" 2
Authenticate all communication between the client and
the server.  Only has an effect on the \fBcvs\fR client.
As of this writing, this is only implemented when using
a GSSAPI connection (see node `GSSAPI authenticated\(aq in the CVS manual).
Authentication prevents certain sorts of attacks
involving hijacking the active \fBtcp\fR connection.
Enabling authentication does not enable encryption.
.SP
.IX "RCSBIN, overriding"
.IX "Overriding RCSBIN"
.IP "" 0
\fB-b \fIbindir\fB\fR
.IP "" 2
In \fBcvs\fR 1.9.18 and older, this specified that
\fBrcs\fR programs are in the \fIbindir\fR directory.
Current versions of \fBcvs\fR do not run \fBrcs\fR
programs; for compatibility this option is accepted,
but it does nothing.
.SP
.IX "TMPDIR, environment variable"
.IX "temporary file directory, set via command line"
.IX "temporary file directory, set via environment variable"
.IX "temporary file directory, set via config"
.IX "temporary files, location of"
.IP "" 0
\fB-T \fItempdir\fB\fR
.IP "" 2
Use \fItempdir\fR as the directory where temporary files are
located.
.SP
The \fBcvs\fR client and server store temporary files in a temporary directory.
The path to this temporary directory is set via, in order of precedence:
.SP
.IP "\(bu" 4
The argument to the global \fB-T\fR option.
.SP
.IP "\(bu" 4
The value set for \fBTmpDir\fR in the config file (server only -
see node `config\(aq in the CVS manual).
.SP
.IP "\(bu" 4
The contents of the \fB$TMPDIR\fR environment variable (\fB%TMPDIR%\fR on
Windows - see node `Environment variables\(aq in the CVS manual).
.SP
.IP "\(bu" 4
/tmp
.SP
.SP
Temporary directories should always be specified as an absolute pathname.
When running a CVS client, \fB-T\fR affects only the local process;
specifying \fB-T\fR for the client has no effect on the server and
vice versa.
.SP
.IX "CVSROOT, overriding"
.IX "Overriding CVSROOT"
.IP "" 0
\fB-d \fIcvs_root_directory\fB\fR
.IP "" 2
Use \fIcvs_root_directory\fR as the root directory
pathname of the repository.  Overrides the setting of
the \fB$CVSROOT\fR environment variable.  see node `Repository\(aq in the CVS manual.
.SP
.IX "EDITOR, overriding"
.IX "Overriding EDITOR"
.IP "" 0
\fB-e \fIeditor\fB\fR
.IP "" 2
Use \fIeditor\fR to enter revision log information.  Overrides the
setting of the \fB$CVSEDITOR\fR and \fB$EDITOR\fR
environment variables.  For more information, see
see node `Committing your changes\(aq in the CVS manual.
.SP
.IP "" 0
\fB-f\fR
.IP "" 2
Do not read the \fB~/.cvsrc\fR file.  This
option is most often used because of the
non-orthogonality of the \fBcvs\fR option set.  For
example, the \fBcvs log\fR option \fB-N\fR (turn off
display of tag names) does not have a corresponding
option to turn the display on.  So if you have
\fB-N\fR in the \fB~/.cvsrc\fR entry for \fBlog\fR,
you may need to use \fB-f\fR to show the tag names.
.SP
.IP "" 0
\fB-H\fR
.IP "" 2
.IP "" 0
\fB--help\fR
.IP "" 2
Display usage information about the specified \fBcvs_command\fR
(but do not actually execute the command).  If you don\(aqt specify
a command name, \fBcvs -H\fR displays overall help for
\fBcvs\fR, including a list of other help options.
.SP
.IX "Read-only repository mode"
.IP "" 0
\fB-R\fR
.IP "" 2
Turns on read-only repository mode.  This allows one to check out from a
read-only repository, such as within an anoncvs server, or from a \fBcd-rom\fR
repository.
.SP
Same effect as if the \fBCVSREADONLYFS\fR environment
variable is set. Using \fB-R\fR can also considerably
speed up checkouts over NFS.
.SP
.IX "Read-only mode"
.IP "" 0
\fB-n\fR
.IP "" 2
Do not change any files.  Attempt to execute the
\fBcvs_command\fR, but only to issue reports; do not remove,
update, or merge any existing files, or create any new files.
.SP
Note that \fBcvs\fR will not necessarily produce exactly
the same output as without \fB-n\fR.  In some cases
the output will be the same, but in other cases
\fBcvs\fR will skip some of the processing that would
have been required to produce the exact same output.
.SP
.IP "" 0
\fB-Q\fR
.IP "" 2
Cause the command to be really quiet; the command will only
generate output for serious problems.
.SP
.IP "" 0
\fB-q\fR
.IP "" 2
Cause the command to be somewhat quiet; informational messages,
such as reports of recursion through subdirectories, are
suppressed.
.SP
.IX "Read-only files, and -r"
.IP "" 0
\fB-r\fR
.IP "" 2
Make new working files read-only.  Same effect
as if the \fB$CVSREAD\fR environment variable is set
(see node `Environment variables\(aq in the CVS manual).  The default is to
make working files writable, unless watches are on
(see node `Watches\(aq in the CVS manual).
.SP
.IP "" 0
\fB-s \fIvariable\fB=\fIvalue\fB\fR
.IP "" 2
Set a user variable (see node `Variables\(aq in the CVS manual).
.SP
.IX "Trace"
.IP "" 0
\fB-t\fR
.IP "" 2
Trace program execution; display messages showing the steps of
\fBcvs\fR activity.  Particularly useful with \fB-n\fR to explore the
potential impact of an unfamiliar command.
.SP
.IP "" 0
\fB-v\fR
.IP "" 2
.IP "" 0
\fB--version\fR
.IP "" 2
Display version and copyright information for \fBcvs\fR.
.SP
.IX "CVSREAD, overriding"
.IX "Overriding CVSREAD"
.IP "" 0
\fB-w\fR
.IP "" 2
Make new working files read-write.  Overrides the
setting of the \fB$CVSREAD\fR environment variable.
Files are created read-write by default, unless \fB$CVSREAD\fR is
set or \fB-r\fR is given.
.SP
.IP "" 0
\fB-x\fR
.IP "" 2
.IX "Encryption"
Encrypt all communication between the client and the
server.  Only has an effect on the \fBcvs\fR client.  As
of this writing, this is only implemented when using a
GSSAPI connection (see node `GSSAPI authenticated\(aq in the CVS manual) or a
Kerberos connection (see node `Kerberos authenticated\(aq in the CVS manual).
Enabling encryption implies that message traffic is
also authenticated.  Encryption support is not
available by default; it must be enabled using a
special configure option, \fB--enable-encryption\fR,
when you build \fBcvs\fR.
.SP
.IP "" 0
\fB-z \fIlevel\fB\fR
.IP "" 2
.IX "Compression"
.IX "Gzip"
Request compression \fIlevel\fR for network traffic.
\fBcvs\fR interprets \fIlevel\fR identically to the \fBgzip\fR program.
Valid levels are 1 (high speed, low compression) to
9 (low speed, high compression), or 0 to disable
compression (the default).  Data sent to the server will
be compressed at the requested level and the client will request
the server use the same compression level for data returned.  The
server will use the closest level allowed by the server administrator to
compress returned data.  This option only has an effect when passed to
the \fBcvs\fR client.
.SP
.SH "Common options"
.SS "Common command options"
.IX "Common options"
.IX "Right-hand options"
.SP
This section describes the \fBcommand_options\fR that
are available across several \fBcvs\fR commands.  These
options are always given to the right of
\fBcvs_command\fR. Not all
commands support all of these options; each option is
only supported for commands where it makes sense.
However, when a command has one of these options you
can almost always count on the same behavior of the
option as in other commands.  (Other command options,
which are listed with the individual commands, may have
different behavior from one \fBcvs\fR command to the other).
.SP
\fBNote: the \fBhistory\fB command is an exception; it supports
many options that conflict even with these standard options.\fR
.SP
.IX "Dates"
.IX "Time"
.IX "Specifying dates"
.IP "" 0
\fB-D \fIdate_spec\fB\fR
.IP "" 2
Use the most recent revision no later than \fIdate_spec\fR.
\fIdate_spec\fR is a single argument, a date description
specifying a date in the past.
.SP
The specification is \fIsticky\fR when you use it to make a
private copy of a source file; that is, when you get a working
file using \fB-D\fR, \fBcvs\fR records the date you specified, so that
further updates in the same directory will use the same date
(for more information on sticky tags/dates, see node `Sticky tags\(aq in the CVS manual).
.SP
\fB-D\fR is available with the \fBannotate\fR, \fBcheckout\fR,
\fBdiff\fR, \fBexport\fR, \fBhistory\fR, \fBls\fR,
\fBrdiff\fR, \fBrls\fR, \fBrtag\fR, \fBtag\fR, and \fBupdate\fR commands.
(The \fBhistory\fR command uses this option in a
slightly different way; see node `history options\(aq in the CVS manual).
.SP
For a complete description of the date formats accepted by \fBcvs\fR,
see node `Date input formats\(aq in the CVS manual.
.SP
Remember to quote the argument to the \fB-D\fR
flag so that your shell doesn\(aqt interpret spaces as
argument separators.  A command using the \fB-D\fR
flag can look like this:
.SP
.PD 0
.SP
.IP "" 4
$ cvs diff -D "1 hour ago" cvs.texinfo

.PD
.IP "" 2
.SP
.IX "Forcing a tag match"
.IP "" 0
\fB-f\fR
.IP "" 2
When you specify a particular date or tag to \fBcvs\fR commands, they
normally ignore files that do not contain the tag (or did not
exist prior to the date) that you specified.  Use the \fB-f\fR option
if you want files retrieved even when there is no match for the
tag or date.  (The most recent revision of the file
will be used).
.SP
Note that even with \fB-f\fR, a tag that you specify
must exist (that is, in some file, not necessary in
every file).  This is so that \fBcvs\fR will continue to
give an error if you mistype a tag name.
.SP
\fB-f\fR is available with these commands:
\fBannotate\fR, \fBcheckout\fR, \fBexport\fR,
\fBrdiff\fR, \fBrtag\fR, and \fBupdate\fR.
.SP
\fBWARNING:  The \fBcommit\fB and \fBremove\fB
commands also have a
\fB-f\fB option, but it has a different behavior for
those commands.  See see node `commit options\(aq in the CVS manual, and
see node `Removing files\(aq in the CVS manual.\fR
.SP
.IP "" 0
\fB-k \fIkflag\fB\fR
.IP "" 2
Override the default processing of RCS keywords other than
\fB-kb\fR.  see node `Keyword substitution\(aq in the CVS manual, for the meaning of
\fIkflag\fR.  Used with the \fBcheckout\fR and \fBupdate\fR
commands, your \fIkflag\fR specification is
\fIsticky\fR; that is, when you use this option
with a \fBcheckout\fR or \fBupdate\fR command,
\fBcvs\fR associates your selected \fIkflag\fR with any files
it operates on, and continues to use that \fIkflag\fR with future
commands on the same files until you specify otherwise.
.SP
The \fB-k\fR option is available with the \fBadd\fR,
\fBcheckout\fR, \fBdiff\fR, \fBexport\fR, \fBimport\fR,
\fBrdiff\fR, and \fBupdate\fR commands.
.SP
\fBWARNING: Prior to CVS version 1.12.2, the \fB-k\fB flag
overrode the \fB-kb\fB indication for a binary file.  This could
sometimes corrupt binary files.  see node `Merging and keywords\(aq in the CVS manual, for
more.\fR
.SP
.IP "" 0
\fB-l\fR
.IP "" 2
Local; run only in current working directory, rather than
recursing through subdirectories.
.SP
Available with the following commands: \fBannotate\fR, \fBcheckout\fR,
\fBcommit\fR, \fBdiff\fR, \fBedit\fR, \fBeditors\fR, \fBexport\fR,
\fBlog\fR, \fBrdiff\fR, \fBremove\fR, \fBrtag\fR,
\fBstatus\fR, \fBtag\fR, \fBunedit\fR, \fBupdate\fR, \fBwatch\fR,
and \fBwatchers\fR.
.SP
.IX "Editor, avoiding invocation of"
.IX "Avoiding editor invocation"
.IP "" 0
\fB-m \fImessage\fB\fR
.IP "" 2
Use \fImessage\fR as log information, instead of
invoking an editor.
.SP
Available with the following commands: \fBadd\fR,
\fBcommit\fR and \fBimport\fR.
.SP
.IP "" 0
\fB-n\fR
.IP "" 2
Do not run any tag program.  (A program can be
specified to run in the modules
database (see node `modules\(aq in the CVS manual); this option bypasses it).
.SP
\fBNote: this is not the same as the \fBcvs -n\fB
program option, which you can specify to the left of a cvs command!\fR
.SP
Available with the \fBcheckout\fR, \fBcommit\fR, \fBexport\fR,
and \fBrtag\fR commands.
.SP
.IP "" 0
\fB-P\fR
.IP "" 2
Prune empty directories.  See see node `Removing directories\(aq in the CVS manual.
.SP
.IP "" 0
\fB-p\fR
.IP "" 2
Pipe the files retrieved from the repository to standard output,
rather than writing them in the current directory.  Available
with the \fBcheckout\fR and \fBupdate\fR commands.
.SP
.IP "" 0
\fB-R\fR
.IP "" 2
Process directories recursively.  This is the default for all \fBcvs\fR
commands, with the exception of \fBls\fR & \fBrls\fR.
.SP
Available with the following commands: \fBannotate\fR, \fBcheckout\fR,
\fBcommit\fR, \fBdiff\fR, \fBedit\fR, \fBeditors\fR, \fBexport\fR,
\fBls\fR, \fBrdiff\fR, \fBremove\fR, \fBrls\fR, \fBrtag\fR,
\fBstatus\fR, \fBtag\fR, \fBunedit\fR, \fBupdate\fR, \fBwatch\fR,
and \fBwatchers\fR.
.SP
.IP "" 0
\fB-r \fItag\fB\fR
.IP "" 2
.IP "" 0
\fB-r \fItag\fB[:\fIdate\fB]\fR
.IP "" 2
.IX "HEAD, special tag"
.IX "BASE, special tag"
Use the revision specified by the \fItag\fR argument (and the \fIdate\fR
argument for the commands which accept it) instead of the
default \fIhead\fR revision.  As well as arbitrary tags defined
with the \fBtag\fR or \fBrtag\fR command, two special tags are
always available: \fBHEAD\fR refers to the most recent version
available in the repository, and \fBBASE\fR refers to the
revision you last checked out into the current working directory.
.SP
The tag specification is sticky when you use this
with \fBcheckout\fR or \fBupdate\fR to make your own
copy of a file: \fBcvs\fR remembers the tag and continues to use it on
future update commands, until you specify otherwise (for more information
on sticky tags/dates, see node `Sticky tags\(aq in the CVS manual).
.SP
The tag can be either a symbolic or numeric tag, as
described in see node `Tags\(aq in the CVS manual, or the name of a branch, as
described in see node `Branching and merging\(aq in the CVS manual.
When \fItag\fR is the name of a
branch, some commands accept the optional \fIdate\fR argument to specify
the revision as of the given date on the branch.
When a command expects a specific revision,
the name of a branch is interpreted as the most recent
revision on that branch.
.SP
Specifying the \fB-q\fR global option along with the
\fB-r\fR command option is often useful, to suppress
the warning messages when the \fBrcs\fR file
does not contain the specified tag.
.SP
\fBNote: this is not the same as the overall \fBcvs -r\fB option,
which you can specify to the left of a \fBcvs\fB command!\fR
.SP
\fB-r \fItag\fB\fR is available with the \fBcommit\fR and \fBhistory\fR
commands.
.SP
\fB-r \fItag\fB[:\fIdate\fB]\fR is available with the \fBannotate\fR,
\fBcheckout\fR, \fBdiff\fR, \fBexport\fR, \fBrdiff\fR, \fBrtag\fR,

X11/ChangeLog
2006-09-08  John Harper  <jsh@apple.com>

	* Tagged: LEOPARD_SUBMISSION_0_44

2006-09-06  John Harper  <jsh@apple.com>

	* xc/lib/GL/apple/dri_dispatch.h,
	xc/lib/GL/apple/dri_dispatch.defs: removed glPointParameteri[v]
	aliases

	* xc/programs/Xserver/hw/apple/rootless-window.c,
	xc/programs/Xserver/hw/apple/quartz.c,
	xc/programs/Xserver/hw/apple/applewmstr.h,
	xc/programs/Xserver/hw/apple/applewm.h,
	xc/programs/Xserver/hw/apple/applewm.c,
	xc/programs/Xserver/hw/apple/applewm-impl.h,
	xc/programs/Xserver/hw/apple/applewm-impl.c,
	xc/programs/Xserver/hw/apple/X11Controller.m,
	xc/programs/Xserver/hw/apple/X11Application.m: use shared
	version of AppleWM headers

	* make-host-def, build.sh, Makefile: removed Xplugin/quartz-wm
	references; attempt to support x86_64

2006-07-07  John Harper  <jsh@apple.com>

	* Tagged: LEOPARD_SUBMISSION_0_43

	* xc/programs/Xserver/hw/apple/x-hash.c: fix compiler warning

	* xc/programs/xterm/main.c: fix xterm build failure (4607761)

2006-02-12  John Harper  <jsh@apple.com>

	* xc/programs/Xserver/GL/apple/indirect.c
	(glAquaInitVisualConfigs): add stereo visuals (4377346)

2006-01-16  John Harper  <jsh@apple.com>

	* Tagged: LEOPARD_SUBMISSION_0_42

	* xc/programs/Xserver/hw/apple/bundle-main.c: added /sbin and
	/usr/sbin to the default PATH value (4408133)

	* xc/programs/Xserver/hw/apple/quartz.c (do_exec): use same
	default PATH and MANPATH values as bundle-main.c (4408133)

2005-11-29  John Harper  <jsh@apple.com>

	Fix xscreensaver-demo crash and make its cross-process GL
	surfaces work correctly (4356702):

	* xc/programs/Xserver/hw/apple/dri-surface.c: don't compile the
	code to draw on surfaces (that was always disables but was
	causing a crash - I've fixed the crash, but the code is never
	used so best to just remove it from the binary)

2005-10-25  John Harper  <jsh@apple.com>

	* Tagged: LEOPARD_SUBMISSION_0_41

	* xc/lib/xtrans/Xtranssock.c: fix Leopard build failure
	(4222517)

2005-10-24  John Harper  <jsh@apple.com>

	* xc/programs/Xserver/hw/apple/bundle-main.c: fix default
	MANPATH (4312329)

2005-08-29  John Harper  <jsh@apple.com>

	* xc/extras/freetype2/src/base/ftmac.c: merged endian changes
	from 3477131

2005-06-22  John Harper  <jsh@apple.com>

	* xc/extras/Xpm/lib/scan.c, xc/extras/Xpm/lib/create.c: merged
	fix for CAN-2005-0605 from xfree86 cvs (4031381)

2005-05-24  John Harper  <jsh@apple.com>

	* xc/programs/Xserver/hw/apple/bundle-main.c
	(install_ipaddr_source): changed the way we listen for network
	/ hostname changes, hopefully to be more reliable

	* xc/programs/Xserver/hw/apple/X11Application.m (sendEvent):
	change fix for bring-all-to-front not to require private
	headers - hardcode enum value we need :-(  (4075093)

2005-05-23  John Harper  <jsh@apple.com>

	* Makefile, make-host-def, merge-libs.sh,
	xc/config/cf/Imake.cf, xc/config/cf/darwin.cf,
	xc/config/cf/darwinLib.tmpl, xc/include/Xmd.h,
	xc/lib/Xt/TMaction.c, xc/lib/Xt/Xtos.h,
	xc/lib/freetype2/Imakefile: checked in changes to
	support building fat libraries when ppc64 is in RC_ARCHS
	(3721701)

	* xc/programs/Xserver/hw/apple/X11.pbproj/project.pbxproj:
	bumped short version number to 1.2 so we don't fail to verify

	* xc/programs/Xserver/hw/apple/X11Application.m: handle scroll
	wheel events with delta < 1 correctly (4109092)

2005-05-20  John Harper  <jsh@apple.com>

	* xc/programs/Xserver/hw/apple/rootless.h,
	xc/programs/Xserver/hw/apple/rootless-window.c,
	xc/programs/Xserver/hw/apple/quartz.h,
	xc/programs/Xserver/hw/apple/quartz.c,
	xc/programs/Xserver/hw/apple/X11Application.m: recognize the
	bfCPSOrderAllWindowsForward bit in activation events and
	respond to it appropriately (4075093)

2005-03-09  John Harper  <jsh@apple.com>

	* Tagged: TIGER_SUBMISSION_0_40

	* xc/programs/Xserver/hw/apple/darwin.c
	(DarwinUpdateModifiers): handle AlphaShift (caps lock) as a
	normal modifier, now that we disabled XKB (when merging with
	XFree86 4.4) that's the correct thing to do (4015381)

2005-03-03  John Harper  <jsh@apple.com>

	* Tagged: TIGER_SUBMISSION_0_39

2005-03-01  John Harper  <jsh@apple.com>

	* xc/extras/Xpm/lib/scan.c, xc/extras/Xpm/lib/create.c: more
	Xpm fixes, from
	https://bugs.freedesktop.org/show_bug.cgi?id=1920, attachment
	1909. (4031381)

	* xc/lib/Xpm/Imakefile: doh, forgot Imakefile portion of
	original patch

	* xc/extras/Xpm/lib/scan.c, xc/extras/Xpm/lib/s_popen.c,
	xc/extras/Xpm/lib/parse.c, xc/extras/Xpm/lib/misc.c,
	xc/extras/Xpm/lib/hashtab.c, xc/extras/Xpm/lib/data.c,
	xc/extras/Xpm/lib/create.c, xc/extras/Xpm/lib/XpmI.h,
	xc/extras/Xpm/lib/WrFFrI.c, xc/extras/Xpm/lib/WrFFrBuf.c,
	xc/extras/Xpm/lib/RdFToI.c, xc/extras/Xpm/lib/RdFToBuf.c,
	xc/extras/Xpm/lib/Imakefile, xc/extras/Xpm/lib/CrDatFrI.c,
	xc/extras/Xpm/lib/CrBufFrI.c, xc/extras/Xpm/lib/Attrib.c:
	merged changes from XFree86 HEAD branch (4022757)

2005-02-15  John Harper  <jsh@apple.com>

	* Tagged: TIGER_SUBMISSION_0_38

	* xc/programs/Xserver/hw/apple/quartz-server.c: request that
	deferred updates are disabled. Otherwise we lose 25%
	performance on the "cat file in xterm" test (3985081)

	Fix 3494176 (deadlock while closing OpenGL window):

	* xc/lib/GL/glx/glxext.c: make __glXLock a recursive mutex

	* xc/lib/GL/apple/dri_driver.h, xc/lib/GL/apple/dri_driver.c:
	make the screen mutex a pthread_mutex and make it recursive

2005-01-04  John Harper  <jsh@apple.com>

	* Tagged: TIGER_SUBMISSION_0_37

	* xc/programs/Xserver/hw/apple/X11Controller.m: fixed several
	bugs in the customization dialog for the applications menu
	(3668769)

2004-11-24  John Harper  <jsh@apple.com>

	* Tagged: TIGER_SUBMISSION_0_36_1

	* Makefile: strip-libraries target doesn't strip libXplugin

2004-11-23  John Harper  <jsh@apple.com>

	* Tagged: TIGER_SUBMISSION_0_36

2004-11-16  John Harper  <jsh@apple.com>

	* xc/programs/Xserver/hw/apple/bundle-main.c: added a signal
	handler to catch SIGTERM etc and exit cleanly, killing the X
	server and xinit (3879878)


tcl/tcl_ext/tcllib/tcllib/modules/doctools/cvs.tcl
# cvs.tcl --
#
#	Handling of various cvs output formats.
#
# Copyright (c) 2003 Andreas Kupries <andreas_kupries@sourceforge.net>
#
# See the file "license.terms" for information on usage and redistribution
# of this file, and for a DISCLAIMER OF ALL WARRANTIES.
# 
# RCS: @(#) $Id: cvs.tcl,v 1.4 2004/01/15 06:36:12 andreas_kupries Exp $

package require Tcl 8.2
package require textutil

namespace eval ::doctools {}
namespace eval ::doctools::cvs {
    namespace export scanLog toChangeLog
}

# ::doctools::cvs::scanLog --
#
#	Scan a log generated by 'cvs log' and extract the relevant information.
#
# Arguments:
#	text	The text to scan
#
# Results:
#	None.
#
# Sideeffects:
#	None.
#
# Notes:
#	Original location of code:	http://wiki.tcl.tk/3638
#				aka	http://wiki.tcl.tk/log2changelog
#	Original author unknown.
#	Bugfix by TR / Torsten Reincke

proc ::doctools::cvs::scanLog {text evar cvar fvar} {

    set text [split $text \n]
    set n    [llength $text]

    upvar $evar entries  ;    #set       entries  [list]
    upvar $cvar comments ;    #array set comments {}
    upvar $fvar files    ;    #array set files    {}

    for {set i 0} {$i < $n} {incr i} {
	set line [lindex $text $i]
	switch -glob -- $line {
	    "*Working file:*" {
		regexp {Working file: (.*)} $line -> filename
	    }
	    "date:*" {
		scan $line "date: %s %s author: %s" date time author
		set author [string trim $author ";"]

		# read the comment lines following date
		set comment ""
		incr i
		set line [lindex $text $i]
		# [TR]: use regexp here to see if log ends:
		while {(![regexp "(-----*)|(=====*)" $line]) && ($i < $n)} {
		    append comment $line "\n"
		    incr i
		    set line [lindex $text $i]
		}

		#  Store this date/author/comment
		lappend entries [list $date $author]
		lappend comments($date,$author) $comment
		lappend files($date,$author,$comment) $filename
	    }
	}
    }

    return
}


# ::doctools::cvs::toChangeLog --

#	Convert a preprocessed cvs log (see scanLog) into a Changelog
#	suitable for emacs.
#
# Arguments:
#	evar, cvar, fvar: Name of the variables containing the preprocessed log.
#
# Results:
#	A string containing a properly formatted ChangeLog.
#
# Sideeffects:
#	None.
#
# Notes:
#	Original location of code:	http://wiki.tcl.tk/3638
#				aka	http://wiki.tcl.tk/log2changelog
#	Original author unknown.

proc ::doctools::cvs::toChangeLog {evar cvar fvar} {
    upvar $evar entries
    upvar $cvar comments
    upvar $fvar files

    set linebuffer [list]

    foreach e [lsort -unique -decreasing $entries] {

	#  print the date/author
	foreach {date author} $e {break}
	lappend linebuffer "$date $author"
	lappend linebuffer ""

	#  Find all the comments submitted this date/author

	set clist [lsort -unique $comments($date,$author)]

	foreach c $clist {
	    #  Print all files for a given comment
	    foreach f [lsort -unique $files($date,$author,$c)] {
		lappend linebuffer "\t* $f:"
	    }

	    #  Format and print the comment

	    lappend linebuffer [textutil::indent [textutil::undent $c] "\t  "]
	    lappend linebuffer ""
	    continue
	}
    }

    return [join $linebuffer \n]
}

#------------------------------------
# Module initialization

package provide doctools::cvs 0.1
